### DISCUSSION TITLE
Watchdog tests: website availability checks

### DISCUSSION THREAD

**Discussion by moving-bits** [cgeo/cgeo#14235] [Q&A]
We use website availability tests as part of our watchdog tasks to regularly check for availability of certain geocaching sites.

A couple of questions arise:

- Do website availability tests on `master` and on `release` check the same? If so: Do we need this?
- Currently they run every four hours, but tests on `master` and on `release` run within half an hour - intended?
- Can we exclude the website availability part from our software builds tests, so that a site which is offline (like GeoKrety yesterday) does not break PR or commit builds?

**Comment by fm-sys** [comment]
The watchdog tests are especially helpful to detect website changes on GC quickly. This should stay as prominent as possible IMHO. 

One solution I could think of:
Check (for minor geocaching Websites) whether the website is available at all and only execute the tests if this is the case.

Our watchdog tests are the same as our normal unit tests AFAIK. We definitely need to execute those for PR builds, to detect unwanted side effects breaking cgeo's core functionality.

**Reply by moving-bits** [reply]
Ok, maybe I should rephrase it: With "watchdog tests" I meant the "website availability" tests only.
All other tests should stay in PR/commit build tests.
Have renamed the discussion and clarified the opening post.
